File: src/main/java/de/mm/fileshuffler/filter/MovieFilter.java
package de.mm.fileshuffler.filter;

import java.io.File;
import java.io.FileFilter;

public class MovieFilter implements FileFilter {
	
	@Override
	public boolean accept(File path) {
		
		if (path.isDirectory()) {
			return true;
		}
		
		String[] parts = path.getPath().split("\\.");
		String ending = parts[parts.length - 1];
		
		switch (ending) {
			case "mkv":
			case "mp4":
			case "avi":
			case "wmv":
			case "mov":
			case "mpg":
			case "mpg2":
				return true;
			default:
				return false;
		}
	}
	
	@Override
	public String toString() {
		return "MovieFilter";
	}
}
 

File: src/main/kotlin/de/mm/fileshuffler/filter/MovieFilter.kt
package de.mm.fileshuffler.filter

import java.io.File
import java.io.FileFilter

class MovieFilter : FileFilter {

	override fun accept(path: File): Boolean {

		if (path.isDirectory) {
			return true
		}

		val parts = path.path.split("\\.".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
		val ending = parts[parts.size - 1]

		when (ending) {
			"mkv", "mp4", "avi", "wmv", "mov", "mpg", "mpg2" -> return true
			else -> return false
		}
	}

	override fun toString(): String {
		return "MovieFilter"
	}
}
 